### PR TITLE
ath79: Add support for Plasma Cloud PA300(E)

### PR DESCRIPTION
**Pull request by ecsv**
* Check the separate patches for the Forwarded: information
* Two devices are in this PR do avoid the conflicts between two independent PRs which depend on the same base modifications

- [x] must be flashable from vendor firmware
  - [ ] webinterface
  - [ ] tftp
  - [x] other: factory image with ap51-flash
- [x] must support upgrade mechanism
  - [x] must have working sysupgrade
    - [x] must keep/forget configuration (if applicable)
      *think `sysupgrade [-n]` or `firstboot`*
  - [x] must have working autoupdate
    *usually means: gluon profile name must match image name*
- [x] reset/wps/phone button must return device into config mode
- [x] primary mac should match address on device label (or packaging) (https://gluon.readthedocs.io/en/latest/dev/hardware.html#notes)
- wired network
  - [x] should support all network ports on the device
  - [x] must have correct port assignment (WAN/LAN)
- wifi (if applicable)
  - [x] association with AP must be possible on all radios
  - [x] association with 802.11s mesh must be working on all radios 
  - [x] ap/mesh mode must work in parallel on all radios
- led mapping
  - power/sys led (_critical, because led definitions are setup on firstboot only_)
    - [x] lit while the device is on
    - [x] should display config mode blink sequence 
(https://gluon.readthedocs.io/en/latest/features/configmode.html)
  - radio leds (**none available**)
    - [ ] should map to their respective radio
    - [ ] should show activity
  - switchport leds (**none available**)
    - [ ] should map to their respective port (or switch, if only one led present) 
    - [ ] should show link state and activity
- outdoor devices only
  - [x] added board name to `is_outdoor_device` function in `package/gluon-core/luasrc/usr/lib/lua/gluon/platform.lua`

### CODE REVIEW

**Review by mweinelt — COMMENTED** [review]
The gluon side of things LGTM. I cannot speak to backporting these patches and would like to defer to @blocktrron or @NeoRaider on those.

File: package/gluon-core/luasrc/lib/gluon/upgrade/010-primary-mac [hunk 1]
```diff
@@ -125,6 +125,10 @@ local primary_addrs = {
 			'om2pv4',
 			'om2p-hsv4',
 		}},
+		{'ath79', 'generic', {
+			'plasmacloud,pa300',
+			'plasmacloud,pa300e',
+		}},
```
Comment (blocktrron) [inline package/gluon-core/luasrc/lib/gluon/upgrade/010-primary-mac:131]: I've just merged #2170 - As both device-trees define the label-mac-device, this is now obsolete.

**Review by blocktrron — COMMENTED** [review]
Same goes for the other Plasma Cloud device. IMHO a backport is fine as long as these boards are already supported in OpenWrt master.

File: package/gluon-core/luasrc/lib/gluon/upgrade/010-primary-mac (hunk 1 above)
Comment (ecsv) [reply package/gluon-core/luasrc/lib/gluon/upgrade/010-primary-mac]: thanks for the info. I have now dropped it.

**Pull request merged**